tfsdk/attribute.go

```diff
@@ -341,18 +341,34 @@ func (a Attribute) validate(ctx context.Context, req ValidateAttributeRequest, r
 				}
 			}
 		case NestingModeSingle:
-			for nestedName, nestedAttr := range a.Attributes.GetAttributes() {
-				nestedAttrReq := ValidateAttributeRequest{
-					AttributePath: req.AttributePath.WithAttributeName(nestedName),
-					Config:        req.Config,
-				}
-				nestedAttrResp := &ValidateAttributeResponse{
-					Diagnostics: resp.Diagnostics,
-				}
+			o, ok := req.AttributeConfig.(types.Object)
 
-				nestedAttr.validate(ctx, nestedAttrReq, nestedAttrResp)
+			if !ok {
+				err := fmt.Errorf("unknown attribute value type (%T) for nesting mode (%T) at path: %s", req.AttributeConfig, nm, req.AttributePath)
+				resp.Diagnostics = append(resp.Diagnostics, &tfprotov6.Diagnostic{
+					Severity:  tfprotov6.DiagnosticSeverityError,
+					Summary:   "Attribute Validation Error",
+					Detail:    "Attribute validation cannot walk schema. Report this to the provider developer:\n\n" + err.Error(),
+					Attribute: req.AttributePath,
+				})
+
+				return
+			}
 
-				resp.Diagnostics = nestedAttrResp.Diagnostics
+			if len(o.Attrs) > 0 {
```

Comment:
Is there a reason this became necessary?

The for loop should cover this case, by just not entering the for loop, which it looks like is what this does?

Unless this is to cover the case where the schema has attributes defined, but Terraform sends a value omitting those attributes, which as far as I know, should never happen?

Reply:
Correct, I'll remove the _if_.

Reply:
I think, as @kmoe mentions in https://github.com/hashicorp/terraform-plugin-framework/issues/112#issuecomment-905493498, it's the call to `tftypes.WalkAttributePath` in `Config.GetAttribute` called from `Attribute.validate` on the nested `Null` Attribute value that fails.

Reply:
Could we use `o.Null` then, instead of checking the number of attributes?

Reply:
Let me take a look...

Reply:
Yes `!o.Null` works just as well.
Commit pushed.

Reply:
will do; thanks